oci/casext/mediatype/parse.go

```diff
@@ -199,4 +214,5 @@ func init() {
 
 	RegisterTarget(ispec.MediaTypeImageManifest)
 	RegisterParser(ispec.MediaTypeImageManifest, manifestParser)
+	RegisterParser(ispec.MediaTypeArtifactManifest, artifactParser)
```

Comment:
This is not enough right?
Currently GC depends on layers:[] array to ensure that they are not deleted.
You also have to now rely on Subject: field.
So let's say I upload an image/artifact, and then a referrer. The original image is now deleted, so the referrer can now be cleaned up.

Reply:
But you an add this a separate PR

Reply:
In this PR it is handled only the situation of finding blobs contained by the notation signature/oci artifact manifest and marking them as reachable in order to avoid deleting blobs that are contained by some oci artifact manifests. 
Also, in order to perform GC on notation signature manifest, in zot this case is handled before calling umoci.GC for blobs, but if we need to cover the gc on oci artifact manifests(checking ref from Subject) in umoci, I could add this in a separate PR.